lib/controllers/users.js

```diff
@@ -1,11 +1,42 @@
 const { Router } = require('express');
+const authenticate = require('../middleware/authenticate');
 const UserService = require('../services/UserService');
 
-module.exports = Router().post('/', async (req, res, next) => {
-  try {
-    const user = await UserService.create(req.body);
-    res.json(user);
-  } catch (e) {
-    next(e);
-  }
-});
+const ONE_DAY_IN_MS = 1000 * 60 * 60 * 24;
+
+module.exports = Router()
+  .post('/', async (req, res, next) => {
+    try {
+      const user = await UserService.create(req.body);
+      res.json(user);
+    } catch (e) {
+      next(e);
+    }
+  })
+  .post('/sessions', async (req, res, next) => {
+    try {
+      const token = await UserService.signIn(req.body);
+      res
+        .cookie(process.env.COOKIE_NAME, token, {
+          httpOnly: true,
+          maxAge: ONE_DAY_IN_MS,
+        })
+        .json({ message: 'Signed in successfully!' });
+    } catch (e) {
+      next(e);
+    }
+  })
+  .get('/me', authenticate, (req, res) => {
+    res.json(req.user);
+  })
+
+  .delete('/sessions', (req, res) => {
+    res
+      .clearCookie(process.env.COOKIE_NAME, {
+        httpOnly: true,
+
+        maxAge: ONE_DAY_IN_MS,
+      })
+      .status(204)
+      .send();
```

Comment:
Perhaps instead of sending a status 204, we could send a message ('Signed out successfully')